packages/consensus/src/validation/script.cairo

```diff
@@ -59,21 +66,44 @@ impl EngineTransactionDummyImpl of EngineTransactionTrait<Transaction, TxIn, TxO
 
 fn script_flags(header: @Header, tx: @Transaction) -> u32 {
     let mut script_flags = 0_u32;
+    let block_height = tx.inputs[0].previous_output.block_height;
 
-    if true { // TODO: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1208
+    // Blocks created after the BIP0016 activation time need to have the
+    // pay-to-script-hash checks enabled.
+    if block_height >= @BIP_16_BLOCK_HEIGHT {
```

Comment:
Why block_height base condition? btcd uses timestamp: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1096

Reply:
That is probably to handle both mainnet & testnet?
In bitcoin core it's enabled for all blocks except for one exception in mainnet
https://github.com/bitcoin/bitcoin/blob/6a370435526875a441001be8c44c9b94a2361c8c/src/validation.cpp#L2368

Reply:
I felt block_height will be more consistent. should I use timestamps instead?

Reply:
> That is probably to handle both mainnet & testnet? In bitcoin core it's enabled for all blocks except for one in mainnet https://github.com/bitcoin/bitcoin/blob/6a370435526875a441001be8c44c9b94a2361c8c/src/validation.cpp#L2368

If we want it to work not only with mainnet at the top level we need an extra parameter descripting network we are validating, right?

Reply:
> I felt block_height will be more consistent. should I use timestamps instead?

I think that, unless we decide to support testnets block_height will do for now.